nucliadb_node/src/services/mod.rs

```diff
@@ -23,12 +23,42 @@ mod versions;
 pub mod reader;
 pub mod writer;
 
-mod shard_disk_structure {
+/// Files and directories hierarchy
+///
+/// DATA_PATH
+/// |-- SHARDS_DIR
+///     |-- <shard-uuid>
+///         |-- PARAGRAPHS_DIR
+///         |-- RELATIONS_DIR
+///         |-- TEXTS_DIR
+///         |-- VECTORSET_DIR
+///         |-- VECTORS_DIR
+pub mod shard_disk_structure {
+    use std::path::{Path, PathBuf};
+
     pub const VERSION_FILE: &str = "versions.json";
-    pub const VECTORS_DIR: &str = "vectors";
-    pub const VECTORSET_DIR: &str = "vectorset";
-    pub const TEXTS_DIR: &str = "text";
+    pub const METADATA_FILE: &str = "metadata.json";
+
+    pub const SHARDS_DIR: &str = "shards";
+
     pub const PARAGRAPHS_DIR: &str = "paragraph";
     pub const RELATIONS_DIR: &str = "relations";
-    pub const METADATA_FILE: &str = "metadata.json";
+    pub const TEXTS_DIR: &str = "text";
+    pub const VECTORSET_DIR: &str = "vectorset";
+    pub const VECTORS_DIR: &str = "vectors";
+
+    /// Path to index node metadata file
+    pub fn metadata_path(data_path: &Path) -> PathBuf {
+        data_path.join(METADATA_FILE)
+    }
+
+    /// Path where all shards are stored
+    pub fn shards_path(data_path: &Path) -> PathBuf {
+        data_path.join(SHARDS_DIR)
+    }
+
+    /// Path where shard `id` is stored
+    pub fn shard_path_by_id(data_path: &Path, id: &str) -> PathBuf {
+        shards_path(data_path).join(id)
+    }
```

Comment:
Why not add this to `SettingsProvider`? We can built those once we have the data path and we would reduce code duplication (this paths are also used in the writer) and improve in code modularity.

Reply:
Even though this are derived from `data_path` and can be seen as configuration, all this values belong to index node's disk organization. Providing this as settings means users of multiple of this settings need to receive either a `Settings` instance or multiple settings.

Get this as an example: https://github.com/nuclia/nucliadb/blob/main/nucliadb_node/src/writer/mod.rs `NodeWriterService` or it's new replacement `UnboundedShardCache` would need to know about `data_path`, `shards_path` and `shard_path_by_id`. Passing all this to the `new` or passing a `Settings` instance seems too much